sql/core/src/main/scala/org/apache/spark/sql/execution/RemoveRedundantProjects.scala

```diff
@@ -97,6 +97,7 @@ object RemoveRedundantProjects extends Rule[SparkPlan] {
       // If a DataSourceV2ScanExec node does not support columnar, a ProjectExec node is required
       // to convert the rows to UnsafeRow. See DataSourceV2Strategy for more details.
       case d: DataSourceV2ScanExecBase if !d.supportsColumnar => false
+      case FilterExec(_, d: DataSourceV2ScanExecBase) if !d.supportsColumnar => false
```

Comment:
Can we add a comment here to explain the `PhysicalOperation` pattern?

Reply:
The comment above covers this case right?

Reply:
@sarutak  is this fix assuming columnar is off on v2 table? what if that is on.

Reply:
@sarutak what if iceberg turns on columnar, does this fix still work?